src/rabbit_connection_tracking.erl

```diff
@@ -0,0 +1,311 @@
+                 end
+               end) of
+                 {atomic,  Val}     -> Val;
+                 {aborted, _Reason} -> 0
+            end
+          catch _ -> 0
+          end
+        end, rabbit_mnesia:cluster_nodes(running)),
+      lists:foldl(fun(X, Acc) -> Acc + X end, 0, Ns)
+    catch
+        _:Err  ->
+            rabbit_log:error(
+              "Failed to fetch number of connections in vhost ~p:~n~p~n",
+              [VirtualHost, Err]),
+            0
+    end.
+
+%% Returns a #tracked_connection from connection_created
+%% event details.
+%%
+%% @see rabbit_connection_tracking_handler.
+tracked_connection_from_connection_created(EventDetails) ->
+    %% Example event:
+    %%
+    %% [{type,network},
+    %%  {pid,<0.329.0>},
+    %%  {name,<<"127.0.0.1:60998 -> 127.0.0.1:5672">>},
+    %%  {port,5672},
+    %%  {peer_port,60998},
+    %%  {host,{0,0,0,0,0,65535,32512,1}},
+    %%  {peer_host,{0,0,0,0,0,65535,32512,1}},
+    %%  {ssl,false},
+    %%  {peer_cert_subject,''},
+    %%  {peer_cert_issuer,''},
+    %%  {peer_cert_validity,''},
+    %%  {auth_mechanism,<<"PLAIN">>},
+    %%  {ssl_protocol,''},
+    %%  {ssl_key_exchange,''},
+    %%  {ssl_cipher,''},
+    %%  {ssl_hash,''},
+    %%  {protocol,{0,9,1}},
+    %%  {user,<<"guest">>},
+    %%  {vhost,<<"/">>},
+    %%  {timeout,14},
+    %%  {frame_max,131072},
+    %%  {channel_max,65535},
+    %%  {client_properties,
+    %%      [{<<"capabilities">>,table,
+    %%        [{<<"publisher_confirms">>,bool,true},
+    %%         {<<"consumer_cancel_notify">>,bool,true},
+    %%         {<<"exchange_exchange_bindings">>,bool,true},
+    %%         {<<"basic.nack">>,bool,true},
+    %%         {<<"connection.blocked">>,bool,true},
+    %%         {<<"authentication_failure_close">>,bool,true}]},
+    %%       {<<"product">>,longstr,<<"Bunny">>},
+    %%       {<<"platform">>,longstr,
+    %%        <<"ruby 2.3.0p0 (2015-12-25 revision 53290) [x86_64-darwin15]">>},
+    %%       {<<"version">>,longstr,<<"2.3.0.pre">>},
+    %%       {<<"information">>,longstr,
+    %%        <<"http://rubybunny.info">>}]},
+    %%  {connected_at,1453214290847}]
+    Name = proplists:get_value(name, EventDetails),
+    Node = proplists:get_value(node, EventDetails),
+    #tracked_connection{id           = {Node, Name},
+                        name         = Name,
+                        node         = Node,
+                        vhost        = proplists:get_value(vhost, EventDetails),
+                        username     = proplists:get_value(user, EventDetails),
+                        connected_at = proplists:get_value(connected_at, EventDetails),
+                        pid          = proplists:get_value(pid, EventDetails),
+                        peer_host    = proplists:get_value(peer_host, EventDetails),
+                        peer_port    = proplists:get_value(peer_port, EventDetails)}.
```

Comment:
this might not lie on the critical path, but use of `proplists:get_value/2` seems quite extensive. how about using `rabbit_misc:pget/2` instead, considering that it was optimised in https://github.com/rabbitmq/rabbitmq-common/pull/51